File: marketmotormicroservices/producto-service/src/main/java/com/jma/productoservice/proveedor/domain/response/ProveedorResponse.java
package com.jma.productoservice.proveedor.domain.response;


import com.jma.productoservice.proveedor.domain.dto.ProveedorDto;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class ProveedorResponse {

    private List<ProveedorDto> content;
    private int pageNo;
    private int pageSize;
    private long totalElements;
    private int totalPages;
    private boolean last;

}

File: frontend-marketmotor/src/app/models/responseapi/ProveedorResponse.ts
import { Proveedor } from "../dtos/Proveedor";

export class ProveedorResponse{

    content: Proveedor[];
    pageNo: number;
    pageSize: number;
    totalElements: number;
    totalPages: number;
    last: boolean;

    constructor(){
        this.content  = [new Proveedor()]
        this.pageNo = 0;
        this.pageSize = 0;
        this.totalElements = 0;
        this.totalPages = 0;
        this.last = false;
    }

}